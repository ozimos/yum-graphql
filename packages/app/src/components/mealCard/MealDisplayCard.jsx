import React from 'react';
import PropTypes from 'prop-types';

const MealDisplayCard = ({
  id,
  title,
  imageUrl,
  price,
  description,
  ...props }) => {
  const meal = { id, title, imageUrl, price, description };

  return (
    <React.Fragment>
      <div className="card">
        <div className="overlay-container">
          <div className="overlay" >
            <div className="flexbox">
              <span className="long_string">
                {title}
              </span>
              <span>
                &#8358;{price}
              </span>
            </div>
          </div>
          <img src={imageUrl} alt="Meal" className="fluid-img" />
        </div>
        <p className="descrip long_string">
          {description}
        </p>
        {props.addToCollection &&
          <button
            className="btn title-button"
            onClick={() => props.addToCollection(meal)}
          >
            {`Add to ${props.collection}`}
          </button>}
        {props.removeFromCollection &&
          <button
            className="btn title-button"
            onClick={() => props.removeFromCollection(id)}
          >
            {`Remove from ${props.collection}`}
          </button>}
      </div>
    </React.Fragment>
  );
};

MealDisplayCard.defaultProps = {
  addToCollection: null,
  collection: null,
  removeFromCollection: null,
};

MealDisplayCard.propTypes = {
  title: PropTypes.string.isRequired,
  collection: PropTypes.string,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  addToCollection: PropTypes.func,
  removeFromCollection: PropTypes.func,
};

export default MealDisplayCard;

