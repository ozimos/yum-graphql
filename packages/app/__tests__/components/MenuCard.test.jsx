/* global React:false, shallow:false toJson:false */

import MealDisplayCard from '../../src/components/mealCard/MealDisplayCard';
import { meal } from '../__mocks__/mealDataMock';

describe('MealDisplayCard Component', () => {

  it('should render without throwing an error', () => {
    expect(shallow(<MealDisplayCard
      {...meal.data}
      dispatch={jest.fn()}
      addToMenu={jest.fn()}
      removeFromMenu={jest.fn()}
    />)
      .exists(<div className="card" />)).toBe(true);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<MealDisplayCard
      {...meal.data}
      dispatch={jest.fn()}
      addToMenu={jest.fn()}
      removeFromMenu={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

