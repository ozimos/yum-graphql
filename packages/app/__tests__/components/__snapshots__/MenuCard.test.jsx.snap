// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealDisplayCard Component renders correctly 1`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          />
          <span>
            â‚¦
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
      />
    </div>
    <p
      className="descrip long_string"
    />
  </div>
</React.Fragment>
`;
